src/js/plugins/plugin.vendor-fullscreen.js

```diff
@@ -0,0 +1,231 @@
+
+/** @override */
+BookReader.prototype.init = (function(super_) {
+  return function() {
+    super_.call(this);
+
+    if (!BookReader.util.fullscreenAllowed()) {
+      return;
+    }
+    // In fullscreen mode the colorbox and overlay need to be inside the fullscreen element to display properly.
+    BookReader.util.bindFullscreenChangeListener(this, (e) => {
+      e.data.resize();
+      e.data.updateBrClasses();
+      const cboxOverlay = $('#cboxOverlay');
+      const cbox = $('#colorbox');
+      if (BookReader.util.isFullscreenActive()) {
+        // In full screen mode, the colorbox and overlay need
+        // to be children of the fullscreen element to display properly.
+        let $fullscreen = $(BookReader.util.getFullscreenElement());
+        $fullscreen.append(cboxOverlay).append(cbox);
+      } else {
+        // In non-fullscreen mode, the colorbox and overlay need
+        // to be children of the main document body.
+        $(document.body).append(cboxOverlay).append(cbox);
+      }
+    });
+  }
+})(BookReader.prototype.init);
+
+/** 
+ * Start fullscreen mode
+ */
+BookReader.prototype.enterFullWindow = function() {
+  this.refs.$brContainer.css('opacity', 0);
+
+  const windowWidth = $(window).width();
+  if (windowWidth <= this.onePageMinBreakpoint) {
+    this.switchMode(this.constMode1up);
+  }
+
+  this.isVendorFullscreenActive = true;
+  this.updateBrClasses();
+
+  this.resize();
+  this.jumpToIndex(this.currentIndex());
+
+  this.refs.$brContainer.animate({ opacity: 1 }, 400, 'linear');
+
+  this._fullWindowExitHandler = function (e) {
+    if (e.keyCode === 27) this.exitFullScreen();
+  };
+  $(document).on('keyup.' + EVENT_NAMESPACE, this._fullWindowExitHandler);
+};
+
+/** 
+ * Exit from fullscreen mode
+ */
+BookReader.prototype.exitFullWindow = function() {
+  this.refs.$brContainer.css('opacity', 0);
+
+  $(document).off('keyup' + EVENT_NAMESPACE);
+
+  this.isFullscreenActive = false;
+  this.updateBrClasses()
+
+  this.resize();
+  this.refs.$brContainer.animate({ opacity: 1 }, 400, 'linear');
+};
+
+/** 
+ * Returns true if fullscreen mode is enabled
+ *
+ * @returns {boolean}
+ */
+BookReader.prototype.isFullscreen = function() {
+  return BookReader.util.isFullscreenActive() || this.isVendorFullscreenActive;
+};
+
+/** 
+ * Toggle screen
+ */
+BookReader.prototype.toggleFullscreen = function() {
+  if (this.isFullscreen()) {
+    if (BookReader.util.fullscreenAllowed()) {
+      BookReader.util.exitFullscreen();
+    } else {
+      this.exitFullWindow();
+    }
+  } else {
+    if (BookReader.util.fullscreenAllowed()) {
+      BookReader.util.requestFullscreen(this.refs.$br[0]);
+    } else {
+      this.enterFullWindow();
+    }
+  }
+};
+
+
+/**
+ * Returns true if current device is mobile
+ *
+ * @returns {boolean}
+ */
+BookReader.util.isMobile = function() {
+  return (typeof window.orientation !== 'undefined') || (navigator.userAgent.indexOf('IEMobile') !== -1);
+};
```

Comment:
Let's deprecate storing these utils on the BR object:

```suggestion
/**
 * Returns true if current device is mobile
 *
 * @returns {boolean}
 */
export function isMobile() {
  return (typeof window.orientation !== 'undefined') || (navigator.userAgent.indexOf('IEMobile') !== -1);
};
```

Then at the bottom of the file:

```js
/** @deprecated */
BookReader.utils.isMobile = isMobile;
```